File: MarxUp/MarxUp/DataSources/ImagesPreviewDataSorce/TabsCollectionViewDataSource.m
//
//  TabsCollectionViewDataSource.m
//  MarxUp
//
//  Created by Ognyanka Boneva on 16.10.18.
//  Copyright © 2018 Ognyanka Boneva. All rights reserved.
//

#import "TabsCollectionViewDataSource.h"
#import "TabsCollectionViewCell.h"
#import "Constants.h"

@interface TabsCollectionViewDataSource ()

@property (strong, nonatomic) NSArray<NSString *> *tabs;

@end


@implementation TabsCollectionViewDataSource

+ (instancetype)newDataSource {
    TabsCollectionViewDataSource *newDataSource = [TabsCollectionViewDataSource new];
    newDataSource.tabs = @[@"TOP", @"VIRAL", @"LATEST", @"YOUR IMAGES"];
    return newDataSource;
}

- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
    return 1;
}

- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
    return self.tabs.count;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
    TabsCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:IDENTIFIER_TABS_CELL forIndexPath:indexPath];
    cell.title.text = self.tabs[indexPath.item];
    cell.title.textColor = UIColor.orangeColor;
    return cell;
}

- (NSString *)tabAtIndex:(NSInteger)index {
    return self.tabs[index];
}

@end

File: MarxUpUp/MarxUpUp/DataSources/ImagesPreviewDataSource/TabsCollectionViewDataSource.swift
//
//  TabsCollectionViewDataSource.swift
//  MarxUpUp
//
//  Created by Ognyanka Boneva on 12.11.18.
//  Copyright © 2018 Ognyanka Boneva. All rights reserved.
//

import UIKit

class TabsCollectionViewDataSource: NSObject {
    private let tabs = [TabModel(title: "TOP", filter: DataFilter(local: false, sort: ImageSort.top)),
                TabModel(title: "VIRAL", filter: DataFilter(local: false, sort: ImageSort.viral)),
                TabModel(title: "LATEST", filter: DataFilter(local: false, sort: ImageSort.date)),
                TabModel(title: "YOUR IMAGES", filter: DataFilter(local: true, sort: ImageSort.none))]

    var defaultSelectedTabIndex: IndexPath {
        return IndexPath(item: 1, section: 0)
    }

    var selectedTabIndex = IndexPath(item: 0, section: 0)

    func filter(atIndex index: Int) -> DataFilter {
        guard index < tabs.count else {
            return tabs[defaultSelectedTabIndex.item].filter
        }
        return tabs[index].filter
    }
}

extension TabsCollectionViewDataSource: UICollectionViewDataSource {
    func collectionView(_ collectionView: UICollectionView,
                        cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCell(fromClass: TabsCollectionViewCell.self, forIndexPath: indexPath)

        cell.backgroundColor = UIColor.clear
        if indexPath == selectedTabIndex {
            cell.backgroundColor = UIColor.orange.withAlphaComponent(0.1)
        }

        cell.tabLabel.text = tabs[indexPath.item].title
        cell.tabLabel.textColor = UIColor.orange

        return cell
    }

    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection: Int) -> Int {
        return tabs.count
    }

    func numberOfSections(in collectionView: UICollectionView) -> Int {
        return 1
    }
}
